File: 216-CombinationSumIII/C++/main.cpp
class Solution {
private:
    vector<vector<int>> res;

public:
    void appendCombination(int k, int n, 
                           vector<int> combination=vector<int>(),
                           int beg=1) {
        if (k == 0 && n == 0) {
            res.push_back(combination);
            return;
        }
        
        for (int i = beg; i < 10; ++i) {
            if (n - i >= 0) {
                combination.push_back(i);
                appendCombination(k-1, n-i, combination, i+1);
                combination.pop_back();
            }
        }
    }
    
    vector<vector<int>> combinationSum3(int k, int n) {
        appendCombination(k, n);
        return res;
        
    }
};

File: 216-CombinationSumIII/Python/main.py
class Solution:
    def combinationSum3(self, k, n):
        """
        :type k: int
        :type n: int
        :rtype: List[List[int]]
        """
        res = []
        
        def helper(k, n, combination=[], beg=1):            
            if k == 0 and n == 0:
                res.append(combination)
                return
            
            for i in range(beg, 10):
                if n - i >= 0:
                    helper(k-1, n-i, combination + [i], i+1)
        
        helper(k, n)
        return res